### PR TITLE
DEMO-113: [Product] - Card View

### PR DESCRIPTION
**Pull request by damianz5**
https://jira.ez.no/browse/DEMO-113

products 22 03 2018 10 39

### CODE REVIEW

File: app/Resources/views/themes/tastefulplanet/card/product.html.twig [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<div class="product-list-item">
+    {% set url = path(location) %}
+    <a href="{{ url }}">
+        {{ ez_render_field(content, 'image', {
+            parameters: {
+                alias: 'product_card',
+                class: 'img-responsive'
```
Comment (SylvainGuittard) [inline app/Resources/views/themes/tastefulplanet/card/product.html.twig]: Please use `img-fluid` ;)
We are using bootstrap 4

File: app/Resources/views/themes/tastefulplanet/full/product_list.html.twig [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            <div class="row">
+                <div class="col-md-10 offset-md-1">
+                    {{ form_start(form, { 'action': path(location) }) }}
+
+                    <div class="col-md-12 section-search">
+                        <div class="input-group">
+                            {{ form_widget(form.searchText, {'attr': {'class': 'search', 'placeholder': 'Search Recipes...'}}) }}
+                            {{ form_widget(form.save, {'attr': {'class': 'btn btn-search'}}) }}
+                            <div class="filter-button">
+                                <button class="btn btn-filters">{{ 'Filters'|trans }} <span class="fa  fa-filter"></span></button>
```
Comment (SylvainGuittard) [inline app/Resources/views/themes/tastefulplanet/full/product_list.html.twig]: `fa  fa-filter` -> extra-space

File: app/Resources/views/themes/tastefulplanet/full/product_list.html.twig [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                <span>{{ 'What is on our menu'|trans }}</span>
+            </div>
+
+            <div class="row">
+                <div class="col-md-10 offset-md-1">
+                    {{ form_start(form, { 'action': path(location) }) }}
+
+                    <div class="col-md-12 section-search">
+                        <div class="input-group">
+                            {{ form_widget(form.searchText, {'attr': {'class': 'search', 'placeholder': 'Search Recipes...'}}) }}
```
Comment (SylvainGuittard) [inline app/Resources/views/themes/tastefulplanet/full/product_list.html.twig]: Is there a way to make this placeholder translatable?

File: app/Resources/views/themes/tastefulplanet/full/product_list.html.twig [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                                <li>{{ 'Reduce number of keywords to get more results.'|trans }}</li>
+                                <li>{{ 'Change the filters.'|trans }}</li>
+                            </ul>
+                        </div>
+                    </div>
+                {% endif %}
+            </div>
+
+            <div class="container">
+                <div class="col-lg-12 space-bottom text-center">
```
Comment (SylvainGuittard) [inline app/Resources/views/themes/tastefulplanet/full/product_list.html.twig]: Just a question: `col-lg-12` -> why `lg` and not `md`?

File: web/assets/css/product_list.css [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+}
+
+.content-type-product-list .product-items {
+    display: flex;
+    flex-wrap: wrap;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: flex-start;
+}
+
```
Comment (SylvainGuittard) [inline web/assets/css/product_list.css]: extra line

File: web/assets/css/product_list.css [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+
+.content-type-product-list .product-items .product-list-item {
+    width: 283px;
+    padding: 0;
+    margin-bottom: 30px;
+    margin-top: 10px;
+    background: #fff;
+    border: 1px solid #C7C7C7;
```
Comment (SylvainGuittard) [inline web/assets/css/product_list.css]: should be `#c7c7c7`

File: web/assets/css/product_list.css [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+}
+
+.content-type-product-list .product-items .product-list-item:empty {
+    border: 0;
+}
+
```
Comment (SylvainGuittard) [inline web/assets/css/product_list.css]: Extra line

File: app/Resources/views/themes/tastefulplanet/full/product_list.html.twig [hunk 8]
```diff
@@ -0,0 +1,162 @@
+{% extends "@ezdesign/pagelayout.html.twig" %}
+
+{% block page_head %}
+    {% set title = ez_content_name(content) ~ " - " ~ content.contentInfo.publishedDate|localizeddate('short', 'short', app.request.locale) %}
```
Comment (SylvainGuittard) [inline app/Resources/views/themes/tastefulplanet/full/product_list.html.twig]: This is a copy/paste from other full views. Remove the date. There is no need here... and we'll have NovaSEOBundle ;)

File: src/AppBundle/Controller/ProductController.php [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Pagerfanta\Pagerfanta;
+
+/**
+ * A content view controller that runs queries based on the matched view configuration
+ * and support pagination using Pagerfanta as a pagination provider.
+ */
+class ProductController
```
Comment (SylvainGuittard) [inline src/AppBundle/Controller/ProductController.php]: I think we can call this controller `ProductListController`. What do you think?

**Comment by SylvainGuittard** [comment]
I think we can also improve the rendering of the cards, but we can see that later

File: app/Resources/views/themes/tastefulplanet/full/product_list.html.twig (hunk 3 above)
Comment (damianz5) [reply app/Resources/views/themes/tastefulplanet/full/product_list.html.twig]: it is translatable already

File: app/Resources/views/themes/tastefulplanet/full/product_list.html.twig (hunk 4 above)
Comment (damianz5) [reply app/Resources/views/themes/tastefulplanet/full/product_list.html.twig]: It was like that in previous meta repo, fixed

Comment (SylvainGuittard) [reply app/Resources/views/themes/tastefulplanet/full/product_list.html.twig]: Keep it, we'll see later

**Comment by damianz5** [comment]
fixed

**Pull request merged**